internal/version/version.go

```diff
@@ -110,25 +110,21 @@ func (v PatchVersion) String() string {
 
 // ToMinor satisfies Version.ToMinor
 func (v PatchVersion) ToMinor() MinorVersion {
-	splitDebug := strings.Split(string(v), debugSuffix)
-	splitVersion := strings.Split(splitDebug[0], ".")
-	latestPatchFormat := fmt.Sprintf("%s.%s", splitVersion[0], splitVersion[1])
-
-	if len(splitDebug) == 2 {
-		latestPatchFormat += debugSuffix
+	matched := versionPattern.FindStringSubmatch(v.String())
+	if matched == nil {
+		return "" // impossible if created via NewVersion or NewPatchVersion
 	}
-
-	return MinorVersion(latestPatchFormat)
+	return MinorVersion(matched[1] + matched[3])
```

Comment:
I chose to concat always even if only in debug matched[3] is non-empty. It was less than adding lines to avoid the allocation.. not sure if go would optimize concatenation against an empty string anyway, but max releases of envoy would ever be hundreds.